test/integration/adapter.test.js

```diff
@@ -353,6 +374,267 @@ describe('MongooseAdapter', () => {
+    await a.removeFilteredPolicy(undefined, undefined, 2, 'data1')
+    rulesAfter = await CasbinRule.find({})
+    assert.deepEqual(rulesAfter.map(rule => [rule.p_type, rule.v0, rule.v1, rule.v2, rule.v3, rule.v4]), [
+      ['g', 'alice', 'admin', 'domain2', undefined, undefined]])
+    e = await newEnforcer(rbacDenyDomainModel, a)
+    assert.deepEqual(await e.getPolicy(), [])
+
+    await a.removeFilteredPolicy(undefined, 'g')
+    await a.removeFilteredPolicy(undefined, 'p')
+    rulesAfter = await CasbinRule.find({})
+    assert.deepEqual(rulesAfter.map(rule => [rule.p_type, rule.v0, rule.v1, rule.v2, rule.v3, rule.v4]), [])
+
+    e = await newEnforcer(rbacDenyDomainModel, rbacDenyDomainPolicy)
+    await a.savePolicy(e.getModel())
+
+    // Remove 'read' related policy rules via a filter.
+    // Two rules:
+    // {'admin', 'domain1', 'data1', 'read', 'allow'},
+    // {'admin', 'domain2', 'data2', 'read', 'allow'}
+    // are deleted.
+    await a.removeFilteredPolicy(undefined, undefined, 3, 'read')
+    rulesAfter = await CasbinRule.find({})
+    assert.deepEqual(rulesAfter.map(rule => [rule.p_type, rule.v0, rule.v1, rule.v2, rule.v3, rule.v4]), [
+      ['p', 'admin', 'domain1', 'data1', 'write', 'allow'],
+      ['p', 'admin', 'domain2', 'data2', 'write', 'allow'],
+      ['p', 'alice', 'domain2', 'data2', 'write', 'deny'],
+      ['g', 'alice', 'admin', 'domain2', undefined, undefined]])
+    e = await newEnforcer(rbacDenyDomainModel, a)
+    assert.deepEqual(await e.getPolicy(), [
+      ['admin', 'domain1', 'data1', 'write', 'allow'],
+      ['admin', 'domain2', 'data2', 'write', 'allow'],
+      ['alice', 'domain2', 'data2', 'write', 'deny']])
+
+    // Remove 'read' related policy rules via a filter.
+    // One rule:
+    // {'alice', 'domain2', 'data2', 'write', 'deny'},
+    // is deleted.
+    await a.removeFilteredPolicy(undefined, undefined, 4, 'deny')
+    rulesAfter = await CasbinRule.find({})
+    assert.deepEqual(rulesAfter.map(rule => [rule.p_type, rule.v0, rule.v1, rule.v2, rule.v3, rule.v4]), [
+      ['p', 'admin', 'domain1', 'data1', 'write', 'allow'],
+      ['p', 'admin', 'domain2', 'data2', 'write', 'allow'],
+      ['g', 'alice', 'admin', 'domain2', undefined, undefined]])
+    e = await newEnforcer(rbacDenyDomainModel, a)
+    assert.deepEqual(await e.getPolicy(), [
+      ['admin', 'domain1', 'data1', 'write', 'allow'],
+      ['admin', 'domain2', 'data2', 'write', 'allow']])
+  })
+
+  it('SyncedAdapter should fail when connecting to non-replicaset database', async () => {
+    // Create SyncedMongoAdapter
+    try {
+      await createFailingSyncedAdapter()
+    } catch (error) {
+      assert(error.message, 'You must provide Mongo URI with replicaSet attribute to connect with synced adapter!')
+    }
+  })
+
+  it('SetSynced should fail in non-replicaset connection', async () => {
+    // Create SyncedMongoAdapter
+    try {
+      const adapter = await createAdapter()
+      adapter.setSynced(true)
```

Comment:
You need to throw errors here, so test could fail if error was not thrown before. There are few cases:

1. `createAdapter()` works as expected and produces an error. You handle it with catch and make assert on it - ok.
2. `createAdapter()` works unexpected and does not produce an error. It does not go into catch block, meaning, you assert nothing, code just continues to work and exits the test, meaning, everything is ok, but it is not.

Reply:
And yes, this way must be done for each test like this one.

Reply:
Ooops, sorry for pointing in the wrong direction ☺️ 
I forgot that assertion libraries has a helper function called `throws`. For instance, take a look at chai - https://www.chaijs.com/api/assert/#method_throws

Better to use this one to avoid such cases I described

```js
assert.throws(fn, 'Error thrown must have this msg');
assert.throws(fn, /Error thrown must have a msg that matches this/);
assert.throws(fn, ReferenceError);
assert.throws(fn, errorInstance);
assert.throws(fn, ReferenceError, 'Error thrown must be a ReferenceError and have this msg');
assert.throws(fn, errorInstance, 'Error thrown must be the same errorInstance and have this msg');
assert.throws(fn, ReferenceError, /Error thrown must be a ReferenceError and match this/);
assert.throws(fn, errorInstance, /Error thrown must be the same errorInstance and match this/);
```

Reply:
I do not think that `throws` works with async functions, but I might be wrong.

Reply:
I believe it should work. I can say for sure, that tests I'm writing in Jest can understand `expect(async () => doSomething()).toThrowError()`. Could be, chai also understands this and can do (maybe latest version)